### PR TITLE
Updates react to v17

### PR DESCRIPTION
**Pull request by KaiVolland**
## Description

This updates react to v17.

As `enzyme` does not support react 17 (https://github.com/enzymejs/enzyme/issues/2429) some tests are failing.
So this also introduces `react-testing-library`. At least for the failing tests.

## Related issues or pull requests

Please list issues or pull requests that the changes you propose are related to. It does not matter if they are still open and/or unmerged, any link is appreciated.

## Pull request type

Please check the type of change your PR introduces:

- [x] Dependency updates
- [x] Build related changes

## Do you introduce a breaking change?

- [x] No

## Checklist

- [x] I understand and agree that the changes in this PR will be licensed under the [BSD 2-Clause License](https://github.com/geostyler/geostyler/blob/master/)
- [x] I have followed the [guidelines for contributing](https://github.com/geostyler/geostyler/blob/master/CONTRIBUTING.md)
- [x] The proposed change fits to the content of the [code of conduct](https://github.com/geostyler/geostyler/blob/master/CODE_OF_CONDUCT.md)
- [x] I have added or updated tests and documentation, and the test suite passes (run `npm test` locally)

### CODE REVIEW

**Comment by marcjansen** [comment]
Donusers need to upgrade their React as well?

**Comment by KaiVolland** [comment]
> Do users need to upgrade their React as well?

Just tested it with the geostlyer-demo (react 16) and it works as expected. I think its generally not a problem to use react components from different versions in your app. Actually the purpose of react 17 is to enhance the work with multiple react verisons: https://reactjs.org/blog/2020/10/20/react-v17.html#no-new-features

File: jest/setup.js [hunk 1]
```diff
@@ -33,3 +33,17 @@ import Enzyme from 'enzyme';
...
 Enzyme.configure({ adapter: new Adapter() });
+
+Object.defineProperty(window, 'matchMedia', {
+  writable: true,
+  value: jest.fn().mockImplementation(query => ({
+    matches: false,
+    media: query,
+    onchange: null,
+    addListener: jest.fn(), // deprecated
+    removeListener: jest.fn(), // deprecated
```
Comment (jansule) [inline jest/setup.js:44]: should we use these, if they are deprecated?

**Review by jansule — APPROVED** [review]
Very nice!

File: jest/setup.js (hunk 1 above)
Comment (KaiVolland) [reply jest/setup.js:44]: As it is just a mock this might help if some libraries do use theses functions. We ourselves do not use this.

**Review by dnlkoch — APPROVED** [review]

**Pull request merged**